utils/metrics.py

```diff
@@ -19,86 +20,136 @@ def precision(output, target, topk=(1,)):
-            score += num_hits / (i+1.0)
+            score += num_hits / (i + 1.0)
 
-    if not actual:
+    if not targets:
         return 0.0
 
-    return score / min(len(actual), k)
+    return score / min(len(targets), k)
 
 
-def mapk(predicted, actual, topk=(1,)):
+def mean_average_precision(outputs, targets, topk=(1, )):
     """
     Computes the mean average precision at k.
-    This function computes the mean average prescision at k between two lists
+    This function computes the mean average precision at k between two lists
     of lists of items.
     Parameters
     ----------
-    actual : list
-             A list of lists of elements that are to be predicted
-             (order doesn't matter in the lists)
-    predicted : list
+    outputs : list
                 A list of lists of predicted elements
                 (order matters in the lists)
-    k : int, optional
+    targets : list
+             A list of lists of elements that are to be predicted
+             (order doesn't matter in the lists)
+    topk : int, optional
         The maximum number of predicted elements
     Returns
     -------
     score : double
             The mean average precision at k over the input lists
     """
-    maxk = max(topk)
-    _, pred = predicted.topk(maxk, 1, True, True)
+    max_k = max(topk)
+    _, pred = outputs.topk(max_k, 1, True, True)
 
-    actual = actual.data.cpu().numpy().tolist()
+    targets = targets.data.cpu().numpy().tolist()
     actual_list = []
-    for a in actual:
+    for a in targets:
         actual_list.append([a])
-    actual = actual_list
+    targets = actual_list
     pred = pred.tolist()
 
     res = []
     for k in topk:
-        ap = np.mean([apk(a, p, k) for a, p in zip(actual, pred)])
+        ap = np.mean(
+            [average_precision(p, a, k) for a, p in zip(targets, pred)]
+        )
         res.append(ap)
     return res
 
 
-def jaccard(y_true, y_pred):
-    y_true = y_true.detach().cpu().numpy()
-    y_pred = y_pred.detach().cpu().numpy()
-    intersection = (y_true * y_pred).sum()
-    union = y_true.sum() + y_pred.sum() - intersection
-    return (intersection + 1e-15) / (union + 1e-15)
+def dice(outputs, targets, eps: float = 1e-7, activation: str = 'sigmoid'):
+    """
+    Computes the average precision at k.
+    This function computes the average precision at k between two lists of
+    items.
+    Parameters
+    ----------
+    outputs : list
+                A list of predicted elements (order does matter)
+    targets : list
+             A list of elements that are to be predicted (order doesn't matter)
+    eps : float
+        TODO
+    activation: str
+        TODO
+    Returns
+    -------
+    score : double
+            Dice score
+    """
+    if activation is None or activation == "none":
+        activation_fn = lambda x: x
+    elif activation == "sigmoid":
+        activation_fn = torch.nn.Sigmoid()
+    elif activation == "softmax":
```

Comment:
is it better to say softmax2d then? or use `toch.nn.__dict__[name] if name in []`?